flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -244,6 +252,8 @@ public RocksDBKeyedStateBackend(
 		this.kvStateInformation = kvStateInformation;
 
 		this.writeOptions = new WriteOptions().setDisableWAL(true);
+		checkArgument(writeBatchSize >= 0, "Write batch size have to be no negative value.");
```

Comment:
I think we don't need to `checkArgument` when the passed by argument is annotated as `@Nonnegative` since this is already a contract.

Reply:
I think the annotation is only really evaluated in IntelliJ by the inspections. It does not prevent anything during runtime, so this extra check does not hurt, in my opinion.

Reply:
as far as I know, annotation currently can only throw an exception in IntelliJ, and only `notnull` annotation supported currently.

Reply:
Let me confirm, does the `@Nonnull` annotation work out of IntelliJ? In previous review I was also suggested to remove null pointer checker with `@Nonnull` annotation, maybe this a diverge of code style taste and we should make some discussion in ML and mark them down in our code convention?

Another fact is that there're many `@Nonnegative` annotated methods without additional checking, like [HeapPriorityQueueStateSnapshot](https://github.com/apache/flink/blob/e505bef16c2c11250a2420610c1605a4e64b6d10/flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapPriorityQueueStateSnapshot.java#L71), [KeyGroupPartitioner](https://github.com/apache/flink/blob/e505bef16c2c11250a2420610c1605a4e64b6d10/flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyGroupPartitioner.java#L108), etc. It might be good if we align our code style? Wdyt? Thanks.

Reply:
All annotations only help either the IDE inspections (if activated) and something like "spotbugs", which we do not currently use.

Reply:
For whether to use `checkArgument` and `checkNotNull` I would use the following heuristic: Use it when you "enter" an API or set of public classes. After arguments have been checked once, or when they are only passed between tighly collaborating classes (non public, same package) you usually don't check again.

Side note: I think we agreed in the code style to not use `@Nonnull` because that is assumed the default, and only use `@Nullable` when something is actually nullable.

Reply:
Ok, makes sense, thanks for the clarification.